src/V3Task.cpp

```diff
@@ -927,15 +915,23 @@ class TaskVisitor : public AstNVisitor {
             AstVar* portp = VN_CAST(nodep->fvarp(), Var);
             UASSERT_OBJ(portp, nodep, "function without function output variable");
             if (!portp->isFuncReturn()) nodep->v3error("Not marked as function return var");
-            if (portp->isWide()) nodep->v3error("Unsupported: Public functions with return > 64 bits wide. (Make it a output instead.)");
-            if (ftaskNoInline || nodep->dpiExport()) portp->funcReturn(false);  // Converting return to 'outputs'
-            if ((nodep->dpiImport() || nodep->dpiExport())
-                && portp->dtypep()->basicp()
-                && portp->dtypep()->basicp()->keyword().isDpiUnreturnable()) {
-                portp->v3error("DPI function may not return type "
-                               <<portp->basicp()->prettyTypeName()
-                               <<" (IEEE 1800-2017 35.5.5)");
+            if (nodep->dpiImport() || nodep->dpiExport()) {
+                AstBasicDType *bdtypep = portp->dtypep()->basicp();
+                if (!bdtypep->isDpiPrimitive()) {
+                    if (bdtypep->isDpiBitVec() && portp->width() > 32) {
+                        portp->v3error("DPI function may not return a > 32 bits wide type "
+                                       "other than basic types. (Make it a output instead.)");
```

Comment:
"other than basic types.\n"
     + V3Error::warnMore() + "... Suggest make it a output instead?");

similar below

Reply:
Done both as
```
  + V3Error::warnMore() + "... Suggest make it an output argument instead?");
```